controllers/ncgamescontroller.js

```diff
@@ -58,8 +58,39 @@ exports.getUsers = (req, res) => {
     res.send({ users });
   });
 };
-exports.getReviews = (req, res) => {
-  fetchReviews().then((reviews) => {
+exports.getReviews = async (req, res) => {
+  const sort_by = req.query.sort_by || "created_at";
+  const order = req.query.order || "desc";
+  const category = req.query.category;
+  if (order !== "desc" && order !== "asc") {
+    return res.status(400).send({ msg: "please enter asc or desc order" });
+  }
+  const categoriesWhitelistBigArray = await checkCategories();
+  const categoriesWhitelist = categoriesWhitelistBigArray.reduce(
+    (acc, item) => {
+      if (!acc.includes(item.category)) {
+        acc.push(item.category);
+      }
+      return acc;
+    },
+    []
+  );
+  if (category) {
+    if (!categoriesWhitelist.includes(category)) {
+      return res.status(400).send({ msg: "please enter valid category" });
+    }
+  }
+
+  if (
+    sort_by !== "title" &&
+    sort_by !== "designer" &&
+    sort_by !== "owner" &&
+    sort_by !== "votes" &&
+    sort_by !== "created_at"
+  ) {
+    return res.status(400).send({ msg: "please enter valid sort by column" });
+  }
```

Comment:
Everything above here (except extracting the query params out of the request) should be handled in the model. No data validation/conditional logic/error handling should happen here. 

On line 80 and line 91 these should be Promise.rejects happening in the model rather than a response here.